### PR TITLE
Fix most compiler warnings

### PR DESCRIPTION
**Pull request by ties**
Fix most compiler warnings except for some deprecation

  * [x] I have updated the changelog in README.md

### CODE REVIEW

File: src/main/java/net/ripe/rpki/commons/provisioning/x509/ProvisioningIdentityCertificate.java [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class ProvisioningIdentityCertificate extends ProvisioningCertificate imp
 
     private static final long serialVersionUID = 1L;
 
-
+    @SuppressWarnings("this-escape")
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/provisioning/x509/ProvisioningIdentityCertificate.java:18]: Why not make the containing class final?

File: src/main/java/net/ripe/rpki/commons/xml/XStreamXmlSerializerBuilder.java [hunk 2]
```diff
@@ -54,6 +54,7 @@ public static <C> XStreamXmlSerializerBuilder<C> newForgivingXmlSerializerBuilde
         return new XStreamXmlSerializerBuilder<C>(objectType, NOT_STRICT);
     }
 
+    @SuppressWarnings("this-escape")
```
Comment (bjpbakker) [inline src/main/java/net/ripe/rpki/commons/xml/XStreamXmlSerializerBuilder.java:57]: Best to make the class final

File: src/main/java/net/ripe/rpki/commons/provisioning/x509/ProvisioningIdentityCertificate.java (hunk 1 above)
Comment (ties) [reply src/main/java/net/ripe/rpki/commons/provisioning/x509/ProvisioningIdentityCertificate.java]: I did not think off that and just went with the easy route...

File: src/main/java/net/ripe/rpki/commons/xml/XStreamXmlSerializerBuilder.java (hunk 2 above)
Comment (ties) [reply src/main/java/net/ripe/rpki/commons/xml/XStreamXmlSerializerBuilder.java]: I checked the usage in `rpki-core`, made this class final, and did some changes that were possible that way.

**Comment by ties** [comment]
Ok to merge after agreement in DM yesterday.

We will do a release _before_ switching to immutable resource set and after.

**Pull request merged**